oktapam/client/cloud_connection.go

```diff
@@ -0,0 +1,192 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/okta/terraform-provider-oktapam/oktapam/utils"
+	"github.com/tomnomnom/linkheader"
+)
+
+type CloudConnectionDetails struct {
+	AccountId  *string `json:"account_id"`
+	ExternalId *string `json:"external_id"`
+	RoleArn    *string `json:"role_arn"`
+}
+
+type CloudConnection struct {
+	DeletedAt              *string                 `json:"deleted_at,omitempty"`
+	ID                     *string                 `json:"id,omitempty"`
+	Name                   *string                 `json:"name"`
+	Provider               *string                 `json:"provider"`
+	CloudConnectionDetails *CloudConnectionDetails `json:"cloud_connection_details"`
+}
+
+type CloudConnectionsListResponse struct {
+	CloudConnections []CloudConnection `json:"list"`
+}
+
+func (c CloudConnection) Exists() bool {
+	return utils.IsNonEmpty(c.ID) && utils.IsBlank(c.DeletedAt)
+}
+
+func (c CloudConnection) ToResourceMap() map[string]any {
+	m := make(map[string]any, 10) // TODO: check that size again
+
+	if c.Name != nil {
+		m[attributes.Name] = *c.Name
+	}
+	if c.ID != nil {
+		m[attributes.ID] = *c.ID
+	}
+	if c.DeletedAt != nil {
+		m[attributes.DeletedAt] = *c.DeletedAt
+	}
+	m[attributes.CloudConnectionDetails] = c.CloudConnectionDetails
```

Comment:
You need map each attribute inside CloudConnectionDetail to the value. May be create `ToResourceMAp()` for `CloudConnectionDetails`. You can check how Principals are being mapped int the  Security Policy client 
https://github.com/okta/terraform-provider-oktapam/blob/366bb84e2a69a2dfff88d8c69c93440bc7dc6b69/oktapam/client/security_policy.go#L76